### PR TITLE
Add unified driver

### PR DESCRIPTION
**Pull request by pfultz2**
This unifies the tf and onnx drivers into a single executable. This will do verifying and performance reports just like the original driver.

### CODE REVIEW

File: src/driver/main.cpp [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        auto m = c.params(p);
+        p.eval(m);
+        std::cout << p << std::endl;
+    }
+};
+
+struct perf : command<perf>
+{
+    compiler c;
+    unsigned n = 100;
```
Comment (kahmed10) [inline src/driver/main.cpp]: Might be useful to print out what default value of n is set to.

File: src/driver/main.cpp [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+    program load()
+    {
+        program p;
+        if(type.empty())
+        {
+            if(ends_with(file, ".onnx"))
+                type = "onnx";
+            else
+                type = "tf";
```
Comment (kahmed10) [inline src/driver/main.cpp]: tensorflow protobufs default to .pb file extensions, so I think we should do a check on the file for this

File: src/driver/argument_parser.hpp [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    }
+};
+
+struct argument_parser
+{
+    struct argument
+    {
+        std::vector<std::string> flags;
+        std::function<bool(argument_parser&, const std::vector<std::string>&)> action{};
+        std::string type    = "";
```
Comment (kahmed10) [inline src/driver/argument_parser.hpp]: can we rename type to something like file_type or graph_type? The function get_type_name<T>() in line 81 is used elsewhere in our code for a different purpose, and the names might get confusing.

Comment (pfultz2) [reply src/driver/argument_parser.hpp]: This is the type of the variable so we can show it in the `--help` message it is unrelated to files or graphs. I can rename the `type` in the loader class to `file_type` to make it clearer.

File: src/driver/main.cpp (hunk 1 above)
Comment (pfultz2) [reply src/driver/main.cpp]: It prints out the default now with `--help` flag.

**Pull request merged**